sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/DiscoveryImpl.scala

```diff
@@ -87,6 +87,13 @@ class DiscoveryImpl(
       in.protocolMajorVersion,
       in.protocolMinorVersion)
 
+    if (in.devMode && BuildInfo.runtimeVersion.compareTo(in.proxyVersion) > 0) {
```

Comment:
Strictly speaking the version will only be behind in dev-mode and when running using the docker-compose file. But checking the `in.devMode` for extra protection.